src/Control/FunFlow/External/Coordinator.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE StrictData             #-}
+{-# LANGUAGE TemplateHaskell        #-}
+{-# LANGUAGE TypeFamilies           #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
+module Control.FunFlow.External.Coordinator where
+
+import           Control.FunFlow.ContentHashable (ContentHash)
+import           Control.Lens
+import           Control.Monad.IO.Class          (MonadIO, liftIO)
+
+import qualified Data.ByteString                 as B
+import qualified Data.Text                       as T
+import           Data.UUID                       (UUID)
+
+import           Network.HostName
+import           System.Clock                    (TimeSpec)
+
+-- | Identifier for a task
+newtype TaskId = TaskId UUID
+  deriving (Eq, Ord)
+
+-- | Information about an executor capable of running tasks. Currently this
+--   is just a newtype wrapper around hostname.
+newtype Executor = Executor HostName
+
+data TaskDescription = TaskDescription {
+    _tdOutput     :: ContentHash
+  , _tdSerialised :: B.ByteString
```

Comment:
At this point this is a placeholder, right?
As I understand it, in the future this should explain to the `Coordinator` how to actually execute the task.

Reply:
Right. Well, actually, the coordinator doesn't need to know how to execute the tasks - it just needs to hold an appropriate representation of them. So I wouldn't expect the serialised form to change - it can be completely opaque to the coordinator.

Reply:
Okay, so it wouldn't be the coordinator itself that submits the task to the executor, but there would be another entity that calls `popTask` and then uses the `TaskDescription` to tell the executor what to do exactly?

Reply:
Right. This is mostly motivated, again, by doing the simplest thing. In future, I can see that we might want the coordinator to actually act as a _scheduler_, and appropriately distribute tasks. But the easiest thing is to do as Luigi does, and just act as a central task repository that executors can pull tasks from.